geniza/corpus/views.py

```diff
@@ -459,6 +463,67 @@ def get(self, request, *args, **kwargs):
         return JsonResponse(dict(annotation_list), encoder=iiif_utils.AttrDictEncoder)
 
 
+class DocumentMerge(FormView, PermissionRequiredMixin):
+    permission_required = ("corpus.change_document", "corpus.delete_document")
+    form_class = DocumentMergeForm
+    template_name = "admin/corpus/document/merge.html"
+
+    def get_success_url(self):
+        return reverse("admin:corpus_document_change", args=[self.primary_document.id])
+
+    def get_form_kwargs(self):
+        form_kwargs = super(DocumentMerge, self).get_form_kwargs()
+        form_kwargs["document_ids"] = self.document_ids
+        return form_kwargs
+
+    def get_initial(self):
+        # Default to first document selected
+        document_ids = self.request.GET.get("ids", None)
+        if document_ids:
+            self.document_ids = [int(pid) for pid in document_ids.split(",")]
+            # by default, prefer the first record created
+            return {"primary_document": sorted(self.document_ids)[0]}
+        else:
+            self.document_ids = []
+
+    def form_valid(self, form):
+        """Merge the selected documents into the primary document."""
+        primary_doc = form.cleaned_data["primary_document"]
+        self.primary_document = primary_doc
+        rationale = form.cleaned_data["rationale"]
+
+        try:
+            secondary_ids = [
+                doc_id for doc_id in self.document_ids if doc_id != primary_doc.id
+            ]
+            secondary_docs = Document.objects.filter(id__in=secondary_ids)
+
+            # Get document strings before they are merged
+            primary_doc_str = str(primary_doc)
+            secondary_doc_str = ", ".join([str(doc) for doc in secondary_docs])
+
+            # Merge secondary documents into the selected primary document
+            user = getattr(self.request, "user", None)
+            primary_doc.merge_with(secondary_docs, rationale, user=user)
+
+            # Display info about the merge to the user
+            new_doc_link = reverse(
+                "admin:corpus_document_change", args=[primary_doc.id]
+            )
+            messages.success(
+                self.request,
+                mark_safe(
+                    f"Successfully merged document(s) {secondary_doc_str} with {primary_doc_str}."
+                ),
+            )
+
+        # error if person has more than one account, no account
```

Comment:
codecov flags this as not being covered — IDK if because you don't have a test case that triggers it, or if it's not possible to happen with geniza data

Reply:
I was unable to trigger this error. I assumed it would show up if a bad ID was provided. I've removed the try/except block entirely. Stop me if necessary.

Reply:
I was pretty sure from the comment that it wouldn't come up here (looked like it was related to merging people with one or more accounts in mep data model), so if you can't trigger it I take that as confirmation! Glad we were able to remove it and simplify.